backend/src/govsg/services/govsg-callback.service.ts

```diff
@@ -88,8 +103,45 @@ const parseWebhook = async (
   throw new UnexpectedWebhookError('Unexpected webhook body')
 }
 
+const sendPasscodeCreationMessage = async (
+  whatsappId: WhatsAppId,
+  clientId: WhatsAppApiClient
+): Promise<MessageId> => {
+  const templateMessageToSend: WhatsAppTemplateMessageToSend = {
+    recipient: whatsappId,
+    apiClient: clientId,
+    templateName: 'sgc_passcode_generation', // TODO: Un-hardcode this
+    params: [],
+    language: WhatsAppLanguages.english,
+  }
+  const isLocal = config.get('env') === 'development'
+  const passcodeCreationWamid =
+    await WhatsAppService.whatsappClient.sendTemplateMessage(
+      templateMessageToSend,
+      isLocal
+    )
+  return passcodeCreationWamid
+}
+
+const createPasscode = () => {
+  return randomInt(0, Math.pow(10, 4)).toString().padStart(4, '0')
+}
+
+const storePrecreatedPasscode = async (
+  govsgMessageId: GovsgMessage['id'],
```

Comment:
Just to confirm, `govsgMessageId` is the initial message sent to the user? If so, how do we intend to handle the case where we send multiple messages for the different languages?

Reply:
Yes.

To be frank, for the `multiple messages for the different languages`, I would like to think about it afterwards. Otherwise, there are too many concerns to address in this feature.

But as a start, there are a few things to consider:

- Restrict language to at most 1
- Restrict language to at most 2 and 1 of them must be English
- Associate `GovsgVerification` with only 1 of the messages in a certain order of languages

Reply:
Sure, I don't think the paths you suggested would radically change the implementation for this too, thanks!